### PR TITLE
Add ES cluster resize capability

### PR DESCRIPTION
**Pull request by willronchetti**
- Adds `ElasticSearchServiceClient`, a wrapper for `boto3.client('es')`
- Implements `resize_elasticsearch_cluster`, issuing an update to the relevant settings
- Integrated test was performed on staging
- Unit tests mock the boto3 API

### CODE REVIEW

File: dcicutils/es_utils.py [hunk 1]
```diff
@@ -8,6 +10,64 @@
...
+     """
+    DEFAULT_HOT_MASTER_NODE_TYPE = 'c5.large.elasticsearch'
+    DEFAULT_HOT_MASTER_NODE_COUNT = 3
+    DEFAULT_HOT_DATA_NODE_TYPE = 'c5.2xlarge.elasticsearch'
+    DEFAULT_HOT_DATA_NODE_COUNT = 2
+
+    DEFAULT_COLD_MASTER_NODE_TYPE = 't2.small.elasticsearch'  # does not matter
+    DEFAULT_COLD_MASTER_NODE_COUNT = 0
+    DEFAULT_COLD_DATA_NODE_TYPE = 'c5.large.elasticsearch'
+    DEFAULT_COLD_DATA_NODE_COUNT = 2
```
Comment (netsettler) [inline dcicutils/es_utils.py:27]: I don't see these values being used. Should they be the default for something?

File: test/test_es_utils.py [hunk 2]
```diff
@@ -1,7 +1,95 @@
...
+            }
+        }
+
+    @staticmethod
+    def mock_update_es_bad_response(DomainName, ElasticsearchClusterConfig):
+        return {
+            'something_else': {
+                'blah': 403
+            }
+        }
```
Comment (netsettler) [inline test/test_es_utils.py:32]: The real meat of this test is not just "does it return this?" but "is this function called with the right args?" so you need to tell it what args are the ones that should provoke this response.  (Same for the above mocks, I was too lazy to illustrate all of them.)

```suggestion
    @staticmethod
    def make_mock_update_es_bad_response(expected_domain, expected_config):
        def mock_update_es_bad_response(DomainName, ElasticsearchClusterConfig):
            assert expected_domain == DomainName
            assert expected_config == ElasticsearchClusterConfig
            return {
                'something_else': {
                    'blah': 403
                }
            }
```

File: test/test_es_utils.py [hunk 3]
```diff
@@ -1,7 +1,95 @@
...
+                data_node_type='c5.large.elasticsearch',
+                data_node_count=2
+            )
+            assert not success
+
+    def test_elasticsearch_service_client_resize_bad_response(self):
+        """ Tests handling of a badly formatted response. """
+        client = ElasticSearchServiceClient()
+        with mock.patch.object(client.client, 'update_elasticsearch_domain_config',
+                               self.mock_update_es_bad_response):
```
Comment (netsettler) [inline test/test_es_utils.py:70]: `To be useful, the following obviously-untested code would need to be filled out:

``suggestion
                               self.make_mock_update_es_bad_response(
                                  expected_domain='fourfront-newtest',  # <-- is that right?
                                  expected_config=...what should be here?...)):
```

**Review by netsettler — APPROVED** [review]
I don't see any really obvious problems here but I think the testing should be done differently to be meaningful.

**Review by netsettler — COMMENTED** [review]
You also need to update the CHANGELOG.

File: dcicutils/es_utils.py (hunk 1 above)
Comment (willronchetti) [reply dcicutils/es_utils.py:27]: The idea is these could be used in foursight.

**Pull request merged**